### DISCUSSION TITLE
Unable to set static proxy in WSL2

### DISCUSSION THREAD

**Discussion by darkRaspberry** [microsoft/WSL#5763] [General]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.20185.1000]
Your Distribution version: Ubuntu 20.04.1 LTS
Whether the issue is on WSL 2 and/or WSL 1: WSL
```

# Steps to reproduce

Try to connect the windows system with any network. In which internet is served throught static proxy.
Then try to set a static proxy by 
- https://github.com/microsoft/WSL/issues/1570#issuecomment-271531911
- https://askubuntu.com/questions/1121521/how-to-configure-http-proxy-with-authentication-on-ubuntu-wsl-on-windows-10
- `Acquire::http::Proxy "http://proxy.foo.com:8080";` in /etc/apt/apt.conf [Website](https://dev.to/david_j_eddy/windows-10-wsl-ubuntu-1804-proxy-configuration-for-apt-fhi#:~:text=Using%20Windows%20Subsystem%20for%20Linux,Firewalls%20and%20tight%20proxy%20security.&text=proxy%20uses%20a%20*.)
- Some other method like placing a function in .bashrc

**WSL logs**: Do you want log.

#  Expected behavior

Proxy should work fine in WSL2 as in Windows.

# Actual behavior

Proxy doesn't work.

```bash
Err:1 http://security.ubuntu.com/ubuntu focal-security InRelease
  Could not connect to 192.168.49.1:2980 (192.168.49.1). - connect (111: Connection refused)
Err:2 http://archive.ubuntu.com/ubuntu focal InRelease
  Could not connect to 192.168.49.1:2980 (192.168.49.1). - connect (111: Connection refused)
Err:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
  Unable to connect to 192.168.49.1:2980:
Err:4 http://archive.ubuntu.com/ubuntu focal-backports InRelease
  Unable to connect to 192.168.49.1:2980:
Reading package lists... Done
Building dependency tree
Reading state information... Done
All packages are up to date.
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/focal/InRelease  Could not connect to 192.168.49.1:2980 (192.168.49.1). - connect (111: Connection refused)
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/focal-updates/InRelease  Unable to connect to 192.168.49.1:2980:
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/focal-backports/InRelease  Unable to connect to 192.168.49.1:2980:
W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/focal-security/InRelease  Could not connect to 192.168.49.1:2980 (192.168.49.1). - connect (111: Connection refused)
W: Some index files failed to download. They have been ignored, or old ones used instead.
```
Pingging a DNS should work!

```bash
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
^C
--- 8.8.8.8 ping statistics ---
11 packets transmitted, 0 received, 100% packet loss, time 10396ms
```

**Comment by therealkenc** [comment]
Ping and an http proxy are not the same. If you can't ping `8.8.8.8`, that is unrelated to the "steps to reproduce" cited.

What does `nslookup proxy.foo.com` return. If it doesn't resolve, then you can't do `Acquire::http::Proxy "http://proxy.foo.com:8080"`, because it won't be able to get the IP of your http proxy server.

**Comment by darkRaspberry** [comment]
i just used a valid static proxy. which i tested in different device and the proxy was working.


> Ping and an http proxy are not the same. If you can't ping `8.8.8.8`, that is unrelated to the "steps to reproduce" cited.
> 
> What does `nslookup proxy.foo.com` return. If it doesn't resolve, then you can't do `Acquire::http::Proxy "http://proxy.foo.com:8080"`, because it won't be able to get the IP of your http proxy server.

 This is one of the approach. i tried to solve this error. An every time i tried ping and got the same result.

**Comment by therealkenc** [comment]
> An every time i tried ping and got the same result.

An http proxy will not get you ping (a ICMP request is not a HTTP request). Moving to discussion.

**Reply by darkRaspberry** [reply]
You are correct for ping. I have no issue but after setting proxy it doesn't open any IP or any curl download. That's my concern. i tried a static IP as proxy. like 1.25.35.52 (hypothetical). like listed in this [website](https://hidemy.name/en/proxy-list/?maxtime=100#list).  I was too frustrated with proxy not working in WSL so. I just reset my whole distro and fix my connection to use proxyless configuration to connect to website.

Thanks for your concern. you may close this issue/discussion.
